### PR TITLE
Introduce AENSv2 namespace to introduce raw data pointers

### PR DESCRIPTION
**Pull request by hanssv**
This PR is supported by the Æternity Crypto Foundation

### CODE REVIEW

**Review by ghallak — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `Int.mulmod` - combined builtin operation for multiplication and modulus.
 - `Crypto.poseidon` - a ZK/SNARK-friendly hash function (over the BLS12-381 scalar field).
 - `Address.to_bytes` - convert an address to its binary representation (for hashing, etc.).
+- Raw data pointers added to `AENS`. In short this introduces new types
+  `AENS.nameV2` and `AENS.pointeeV2`, where the latter has a constructor
+  `DataPtV2(string)`; `AENS.lookup` and `AENS.update` consume and produce the
+  new types. Standard library `AENSCompat` is added to convert between old and
+  new pointers.
```
Comment (radrow) [inline CHANGELOG.md:17]: I don't like mixing incompatible functionalities in the same namespace. Let's create `AENS2` namespace and pack all the new stuff in there.

**Review by radrow — CHANGES_REQUESTED** [review]

**Comment by radrow** [comment]
Btw, shouldn't there be changes in `aeso_fcode_to_fate`?

**Comment by hanssv** [comment]
I really like the idea of a whole new top-level namespace, I'll test it and see how it turns out.

@radrow there are no changes at the byte code instruction level (yet!) so no changes necessary in `aeso_fcode_to_fate`

**Comment by hanssv** [comment]
Moved all the active stuff + the new data pointer to `AENSv2` - this turned out nicely I think - good call @radrow !

**Review by radrow — APPROVED** [review]
Looks way better

**Pull request merged**